### PR TITLE
Master

### PR DESCRIPTION
**Pull request by djandreski**
Multi drive support

### CODE REVIEW

**Comment by super3** [comment]
@gamedevsam Feedback?

File: app/driveshare.html [hunk 1]
```diff
@@ -308,6 +258,8 @@ <h4 class="modal-title">Console Logs</h4>
 	.script("lib/electron_boilerplate/env_config.js")
 	.script("lib/electron_boilerplate/context_menu.js")
 	.script("lib/electron_boilerplate/external_links.js")
+	/*.script('bower_components/angular/angular.js')
```
Comment (gamedevsam) [inline app/driveshare.html]: Remove commented block of code unless it's part of a change that is about to happen.

File: app/driveshare.html [hunk 2]
```diff
@@ -319,9 +271,9 @@ <h4 class="modal-title">Console Logs</h4>
 			], function(logs, about, userdata, updater, process, setup) {
 				logs.init();
 				about.init();
+				process.init();
```
Comment (gamedevsam) [inline app/driveshare.html]: Why did you change the init order?

File: app/lib/css/style.css [hunk 3]
```diff
@@ -48,7 +48,7 @@ header {
 
 .tab-list {
   margin-bottom: 1em;
-  display: none; }
+  /*display: none;*/ }
```
Comment (gamedevsam) [inline app/lib/css/style.css]: Remove temporary commented code.

File: app/modules/process.js [hunk 4]
```diff
@@ -8,56 +7,22 @@ var os = require('os');
...
 var ipc = require("electron-safe-ipc/guest");
-
-var userData;
-var l = Ladda.create($('#start').get(0));
 var logs = requirejs('./modules/logs');
 
-exports.child;
-exports.currentProcess;
+exports.children = [];
+exports.currentProcess = [];
```
Comment (gamedevsam) [inline app/modules/process.js]: A better name for this would be plural: ```currentProcesses```

File: app/modules/process.js [hunk 5]
```diff
@@ -67,55 +32,49 @@ var bootstrapProcess = function(name, args) {
...
+exports.register = function(dataservClient) {
+	bootstrapProcess(dataservClient, 'REGISTERING', ['register']);
 };
 
-exports.poll = function() {
-	if(userData.hasValidSettings()) {
-		bootstrapProcess('POLLING', ['poll']);
-	}
+exports.poll = function(dataservClient) {
+	//if(userData.hasValidSettings()) {
```
Comment (gamedevsam) [inline app/modules/process.js]: Remove commented out lines of code.

File: app/modules/process.js [hunk 6]
```diff
@@ -137,21 +96,17 @@ exports.validateDataservClient = function(callback) {
 				callback(output);
 			}
 		});
-	} else if(callback) {
-		callback('error: invalid dataserv-client');
-	}
+	//} else if(callback) {
```
Comment (gamedevsam) [inline app/modules/process.js]: All of them.

File: sjcxc address [hunk 7]
```diff
@@ -0,0 +1,3 @@
+them beam trickle radio birth guitar master lonely pie somebody sudden flower
```
Comment (gamedevsam) [inline sjcxc address]: What the heck is this?

**Comment by gamedevsam** [comment]
I made some comments but I need more time to review all the code. Please respond to the comments by making the changes requested and work towards a clean merge.

File: sjcxc address (hunk 7 above)
Comment (djandreski) [inline sjcxc address]: just ignore it, it is a temp coin address I used for testing

File: app/driveshare.html (hunk 1 above)
Comment (djandreski) [inline app/driveshare.html]: ok I will remove it

File: app/driveshare.html (hunk 2 above)
Comment (djandreski) [inline app/driveshare.html]: since the process script is now used by the userdata script, I wanted to be sure that process is going to be bootstrapped before the userdata makes use of it

File: sjcxc address (hunk 7 above)
Comment (gamedevsam) [inline sjcxc address]: The more important question is why is this part of this change? Did you commit a temp test file? You might have to delete it and mark it for removal.

File: app/modules/userdata.js [hunk 8]
```diff
@@ -11,176 +12,270 @@ var app = remote.require('app');
...
+				
+				process.saveConfig(tabData.dataservClient, tabData.payoutAddress);
+				validate(bQuerySJCX, tabData);
+			}
+		});	
+	} catch (error) {
+		console.log(error.toString());
+	}
+	
+	/*for (var i = 0; i < exports.tabs.length; i++) {
```
Comment (gamedevsam) [inline app/modules/userdata.js]: Remove broken commented out code.

File: app/driveshare.html (hunk 2 above)
Comment (gamedevsam) [inline app/driveshare.html]: :+1:

File: app/driveshare.html (hunk 1 above)
Comment (gamedevsam) [inline app/driveshare.html]: You should make the changes, commit them to your branch and push them to your fork of the repo, it should automatically update this pull request.

File: app/driveshare.html [hunk 9]
```diff
@@ -21,72 +21,22 @@ <h2 class="title"><a href="http://driveshare.org/" class="js-external-link">Driv
 					</div>
 				</header>
 
+				<!-- <div ng-app="driveshareApp" ng-controller="drivesController">
```
Comment (gamedevsam) [inline app/driveshare.html]: Remove commented out block of code.

File: settings.json [hunk 10]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (gamedevsam) [inline settings.json]: Please remove this file from this change.

**Comment by gamedevsam** [comment]
You're almost there, just remove the tmp / debug files from the PR and I'll merge it in!

**Comment by super3** [comment]
@gamedevsam Is this ready to merge?

**Comment by gamedevsam** [comment]
Wonderful, thank you very much!

**Comment by gamedevsam** [comment]
We need to test this and make a new release (only update app/package.json version after a new release is available for download, since that's what gets checked by the version checker in the app).
